### ISSUE TITLE
doesn't read stdin on Cygwin

### ISSUE THREAD

**Issue by VladimirAlexiev** [tomnomnom/gron#32]
I got https://github.com/tomnomnom/gron/releases/download/v0.5.1/gron-windows-amd64-0.5.1.zip.
- works ok from `cmd` (though need to specify `-m` to avoid the colors, or use a better terminal like conemu)
- when I run from mintty & bash, doesn't read stdin:
```bash
$ gron http://jsonplaceholder.typicode.com/users/1
# works ok
$ curl -s http://jsonplaceholder.typicode.com/users/1 | gron -
# nothing happens
$ uname
CYGWIN_NT-6.1 
```

**Comment by tomnomnom** [comment]
Hi @VladimirAlexiev! Thanks for raising an issue :)

I'm afraid I don't own a Windows machine and I'm not massively familiar with cygwin so I might struggle to reproduce this.

Can you answer a few things for me?

1. What shell is being used? Is it bash? sh?
2. Does the curl command work by itself? 
3. Does `echo '{"one":1}' | gron` work?
4. How about `gron <(echo '{"one":1}')`?

If that last one works then this might be a workaround for now:

```
▶ gron <(curl -s http://jsonplaceholder.typicode.com/users/1)
json = {};
json.address = {};
json.address.city = "Gwenborough";
json.address.geo = {};
...
```

A did a little bit of digging and found that cygwin and windows binaries can be a little odd, and that this might help: https://github.com/rprichard/winpty - but again I don't have a way to test it.

Sorry I can't be more help! If the winpty thing doesn't resolve the issue then I'll try my best to get a Windows VM up and running so I can debug things more easily.

**Comment by VladimirAlexiev** [comment]
Hi @tomnomnom, thanks for looking into this!

1. bash. I tried bash both in mintty and in cmd, the result is the same
2. yes
3. yes (and I didn't know one can skip `-` but that makes sense)
4. No. I get this error: `open /dev/fd/63: The system cannot find the path specified`.

I read about winpty but can't figure out how it can help because I don't know what is `pty-master`. Says it helps Windows console apps with interactive input: but gron isn't interactive

Do you use STDIN in some unusual way? I got many cygwin programs that handle `|` just fine. Eg this works ok
`curl -s http://jsonplaceholder.typicode.com/users/1 | less`

Cheers!

**Comment by tomnomnom** [comment]
@VladimirAlexiev: no problem :)

I think the strangest thing here is that `echo '{"one":1}' | gron` works, but `curl ... | gron` does not!

Before we go any further, can you give this build a try and see if it works: [gron-go1.10-build.zip](https://github.com/tomnomnom/gron/files/1875004/gron-go1.10-build.zip)?

It might be a little bit of a long-shot, but it's an easy one to test. I found [an issue](https://github.com/golang/go/issues/22207) in the version of Go that was used to build the binary you're using that sounds as though it may be related, so I've built one with a newer version of Go.

Let me know if you have any luck with that, and if not I'll go and get myself a Windows VM :)

**Comment by joekyo** [comment]
This may be related to https://github.com/golang/go/issues/22024
As I tested, when compile gron using go 1.9 and go 1.9.1, this issue can be reproduced.
You can download gron version 0.5.0 which compiled by go 1.8, or compile gron using go 1.9.2 or newer.

**Comment by tomnomnom** [comment]
Thanks @joekyo! The version I posted above was built with Go 1.10, which *should* work too in theory! If it does I'll cut a release built with 1.10 so the latest release works for everyone

**Comment by VladimirAlexiev** [comment]
gron-go1.10-build.zip works fine. Thanks!!!

**Comment by tomnomnom** [comment]
@VladimirAlexiev fantastic! I'll cut a new release now so others don't hit the same issue :)

**Comment by tomnomnom** [comment]
I've released [v0.5.2](https://github.com/tomnomnom/gron/releases/tag/v0.5.2) built with Go 1.10; so all should be well with the world! Thanks again!
